Source/WebCore/dom/TrustedTypeUtils.cpp

```diff
@@ -0,0 +1,170 @@
+
+namespace WebCore {
+using namespace JSC;
+
+struct TrustedTypeVisitor {
+    String operator()(std::nullptr_t)
+    {
+        return nullString();
+    }
+    String operator()(Ref<TrustedHTML> value)
+    {
+        return value->toString();
+    }
+    String operator()(Ref<TrustedScript> value)
+    {
+        return value->toString();
+    }
+    String operator()(Ref<TrustedScriptURL> value)
+    {
+        return value->toString();
+    }
+};
+
+String trustedTypeToString(const TrustedType trustedType)
+{
+    switch (trustedType) {
+    case TrustedType::TrustedHTML:
+        return "TrustedHTML"_s;
+    case TrustedType::TrustedScript:
+        return "TrustedScript"_s;
+    case TrustedType::TrustedScriptURL:
+        return "TrustedScriptURL"_s;
+    default:
+        ASSERT_NOT_REACHED();
+        return nullString();
+    }
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#process-value-with-a-default-policy-algorithm
+std::variant<std::nullptr_t, Ref<TrustedHTML>, Ref<TrustedScript>, Ref<TrustedScriptURL>> processValueWithDefaultPolicy(ScriptExecutionContext& scriptExecutionContext, const TrustedType expectedType, const String& input, const String& sink)
+{
+    VM& vm = scriptExecutionContext.vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    RefPtr<TrustedTypePolicy> protector = nullptr;
+    if (RefPtr document = dynamicDowncast<Document>(scriptExecutionContext)) {
+        if (auto window = document->domWindow()) {
+            auto trustedTypesFactory = WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes(*window);
+            protector = trustedTypesFactory->defaultPolicy();
+        }
+    } else if (RefPtr workerGlobalScope = dynamicDowncast<WorkerGlobalScope>(scriptExecutionContext)) {
+        auto trustedTypesFactory = WindowOrWorkerGlobalScopeTrustedTypes::trustedTypes(*workerGlobalScope);
+        protector = trustedTypesFactory->defaultPolicy();
+    }
+
+    if (!protector)
+        return nullptr;
+
+    auto jsExpectedType = JSC::jsString(vm, trustedTypeToString(expectedType));
+    JSC::Strong<JSC::Unknown> strongExpectedType(vm, jsExpectedType);
+    auto jsSink = JSC::jsString(vm, sink);
+    JSC::Strong<JSC::Unknown> strongSink(vm, jsSink);
+    FixedVector<JSC::Strong<JSC::Unknown>> arguments({ strongExpectedType, strongSink });
+    auto policyValueHolder = protector->getPolicyValue(expectedType, input, WTFMove(arguments), false);
+    if (policyValueHolder.hasException()) {
+        propagateException(*scriptExecutionContext.globalObject(), scope, policyValueHolder.releaseException());
+        return nullptr;
+    }
+
+    auto policyValue = policyValueHolder.releaseReturnValue();
+    if (policyValue.isNull())
+        return nullptr;
+
+    std::variant<std::nullptr_t, Ref<TrustedHTML>, Ref<TrustedScript>, Ref<TrustedScriptURL>> result;
+    if (expectedType == TrustedType::TrustedHTML) {
+        result.emplace<1>(TrustedHTML::create(policyValue));
+        return result;
```

Comment:
Why do we need this dance here?
Can we simply do: `return { TrustedHTML::create(policyValue) }` or something?